### PR TITLE
Drop Python2 and 3.5 (and iron python)

### PR DESCRIPTION
**Pull request by newville**
## Description

drop support for Python 2.7 and 3.5.

### CODE REVIEW

**Comment by newville** [comment]
I would like to merge this soon-ish (next couple of days) so that we can then merge in #214 (which might need a rebase for `dbr.py`).    
This may not be done and/or need tweaking, but I'd like to get this into master before making more aggressive changes to the testing setup.

File: epics/ca.py [hunk 1]
```diff
@@ -26,10 +26,10 @@
 
 from math import log10
 from pkg_resources import resource_filename
+from copy import deepcopy
 
-from .utils import (STR2BYTES, BYTES2STR, NULLCHAR_2,
-                    strjoin, memcopy, is_string, is_string_or_bytes,
-                    ascii_string, clib_search_path)
+from .utils import (str2bytes, bytes2str, strjoin,
```
Comment (klauer) [inline epics/ca.py:31]: Name changes are a step in the right direction, but you may consider back-compat for the old names. Even if you don't consider these part of the public API, I'd bet that some out there are using it:

```
# Back-compat; deprecated and to be dropped in pyepics x.y.z
STR2BYTES = str2bytes
```

**Review by klauer — COMMENTED** [review]
Seems good overall 👍 

Farewell, Python 2!

File: epics/ca.py (hunk 1 above)
Comment (newville) [reply epics/ca.py:31]: @klauer   Yeah, I think you're probably right, I will add back-compatible aliases for all of those legacy names in `utils.py` where they can enjoy a long and boring life now that we know we don't really need them.

**Comment by newville** [comment]
ok, merging this...

**Pull request merged**